inc/class-wp-sitemaps-renderer.php

```diff
@@ -150,7 +150,14 @@ public function get_sitemap_index_xml( $sitemaps ) {
 
 		foreach ( $sitemaps as $entry ) {
```

Comment:
This should check whether `$name` is `loc` or `lastmod` and doing a `_doing_it_wrong()` if not, as in

```
foreach ( $sitemaps as $entry ) {
	$sitemap = $sitemap_index->addChild( 'sitemap' );
	// Add each element as a child node to the URL entry.
	foreach ( $entry as $name => $value ) {
		if ( 'loc' === $name ) {
			$sitemap->addChild( $name, esc_url( $value ) );
		} elseif ( 'lastmod' === $name ) {
			$sitemap->addChild( $name, esc_attr( $value ) );
		} else {
			_doing_it_wrong(
				__METHOD__,
				/* translators: %s: list of element names */
				sprintf(
					__( 'Fields other than %s are not currently supported for sitemap indexes.', 'core-sitemaps' ),
					implode( ',', array( 'sitemap', 'lastmod' ) )
				),
				'5.5.0'
			);
		}
	}
}
```
```